### PR TITLE
Ttv #168871974 admin tspp

### PR DESCRIPTION
**Pull request by donaldthai**
## Description

As a System Admin
I want to be able to see a list of the TSPPs in the system
So I can spot-check TSPP data after a big data load

## Reviewer Notes

- There's a bug (https://github.com/marmelab/admin-on-rest/issues/2) with the datagrid not overflowing the table. I ended up just overflowing the entire sidebar and content until there's an official fix.
- Two filters for the TDL and TSP IDs
- We currently have a bug that the total count doesn't update based on the filters
- The filter labels may be a bit long, should we abbreviate it?

## Setup

```sh
make build
make server_run
make admin_client_run
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168871974) for this change

## Screenshots

image

With filters
image

### CODE REVIEW

**Comment by donaldthai** [comment]
> This is looking good overall. Regarding the overflow issue, I wonder if there is a subset of important columns that we can show in the index and have the show page (when we get to that story) for each record contain all the information for a given TSPP. That way, you don't have to scroll to find the pagination controls.

@garrettqmartin8 sounds like we could remove the column IDs of the TDL and TSP?

File: pkg/handlers/adminapi/transportation_service_provider_performances.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+// IndexTSPPsHandler returns a list of office users via GET /transportation_service_provider_performances
+type IndexTSPPsHandler struct {
+	handlers.HandlerContext
+	services.TransportationServiceProviderPerformanceListFetcher
+	services.NewQueryFilter
+	services.NewPagination
+}
+
+// Handle retrieves a list of office users
```
[the remark this answers is not in the document]
Comment (donaldthai) [reply pkg/handlers/adminapi/transportation_service_provider_performances.go]: went with transportation service provider performance instead :)

File: pkg/handlers/adminapi/transportation_service_provider_performances.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		payload[i] = payloadForTSPPModel(s)
+	}
+
+	return tsppop.NewIndexTSPPsOK().WithContentRange(fmt.Sprintf("tspps %d-%d/%d", pagination.Offset(), pagination.Offset()+queriedTSPPsCount, totalTSPPsCount)).WithPayload(payload)
+}
+
+// generateQueryFilters is helper to convert filter params from a json string
+// of the form `{"traffic_distribution_list_id": "8e4b3caf-98dc-462a-bbcc-1977d08a08eb" "transportation_service_provider_id": "8e4b3caf-98dc-462a-bbcc-1977d08a08eb"}`
+// to an array of services.QueryFilter
+func (h IndexTSPPsHandler) generateQueryFilters(filters *string, logger handlers.Logger) []services.QueryFilter {
```
Comment (garrettqmartin8) [inline pkg/handlers/adminapi/transportation_service_provider_performances.go]: Nice!

Comment (lynzt) [reply pkg/handlers/adminapi/transportation_service_provider_performances.go]: 👍

**Pull request merged**